src/AzureAuth/CommandMain.cs

```diff
@@ -291,13 +305,19 @@ private bool ValidateOptions()
                 validOptions = false;
             }
 
+            if (!LegalFileNameChecker.IsValidFilename(this.CacheFileName))
+            {
+                this.logger.LogError($"The {CacheFileNameOption} field or environment varable {nameof(EnvVars.AZUREAUTH_CACHE_FILE)} is invalid");
+                validOptions = false;
+            }

```

Comment:
I think you've added this check (again) here because we might have pulled the filename from the environment - which will skip the `LegalFileNameAttribute` check. I'd rather not have 2 copies of this check, and instead do all the checking in 1 place - here. This way we don't need to add the attribute class at all. We can still customize the error message in the checking logic to say if the it was the option, or the env var that was used and had an invalid filename.

Reply:
That's true. We shouldn't repeat ourselves.